File: app/src/main/java/myjin/pro/ahoora/myjin/utils/ServerStatus.java
package myjin.pro.ahoora.myjin.utils;

import android.content.Context;
import android.graphics.drawable.Drawable;
import android.support.annotation.NonNull;
import android.util.Log;
import android.widget.Toast;

import java.util.ArrayList;

import myjin.pro.ahoora.myjin.interfaces.TempListener;
import myjin.pro.ahoora.myjin.models.TempModel;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class ServerStatus {

    private TempListener tempListener;
    private Context mContaxt;
    private String v = "";

    public ServerStatus(TempListener tempListener, Context context) {
        this.tempListener = tempListener;
        this.mContaxt = context;

    }

    public void IsOkServer() {
        try {
            KotlinApiClient.INSTANCE.getClient().create(ApiInterface.class).IsOkServer().enqueue(new Callback<TempModel>() {
                @Override
                public void onResponse(@NonNull Call<TempModel> call, @NonNull Response<TempModel> response) {

                    try {
                        TempModel tempModel = response.body();
                        assert tempModel != null;
                        v = tempModel.getVal();
                        Log.e("val", v + "");
                        VarableValues.INSTANCE.setNetworkState(true);
                        if (v.equals("ok")) {
                            tempListener.IsOk();
                        } else {
                            tempListener.IsNotOk();
                        }
                    } catch (Exception e) {
                        VarableValues.INSTANCE.setNetworkState(true);
                        tempListener.IsNotOk();
                    }
                }

                @Override
                public void onFailure(@NonNull Call<TempModel> call, @NonNull Throwable t) {
                    VarableValues.INSTANCE.setNetworkState(false);
                    tempListener.IsNotOk();

                }
            });

        } catch (Exception e) {
            Toast.makeText(mContaxt, "خطایی رخ داده است", Toast.LENGTH_SHORT).show();
            tempListener.IsNotOk();
        }

    }

}

File: app/src/main/java/myjin/pro/ahoora/myjin/utils/ServerStatus.kt
package myjin.pro.ahoora.myjin.utils

import android.content.Context
import android.util.Log
import android.widget.Toast

import myjin.pro.ahoora.myjin.interfaces.ServerStatusResponse
import myjin.pro.ahoora.myjin.models.TempModel
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class ServerStatus(private val serverStatusResponse: ServerStatusResponse, private val mContext: Context) {
    private var v = ""

    fun checkServer() {
        try {
            KotlinApiClient.client.create(ApiInterface::class.java).IsOkServer().enqueue(object : Callback<TempModel> {
                override fun onResponse(call: Call<TempModel>, response: Response<TempModel>) {

                    try {
                        val tempModel = response.body()!!
                        v = tempModel.getVal()
                        Log.e("val", v + "")
                        VarableValues.NetworkState = true
                        if (v == "ok") {
                            serverStatusResponse.isOk()
                        } else {
                            serverStatusResponse.notOk()
                        }
                    } catch (e: Exception) {
                        VarableValues.NetworkState = true
                        serverStatusResponse.notOk()
                    }

                }

                override fun onFailure(call: Call<TempModel>, t: Throwable) {
                    VarableValues.NetworkState = false
                    serverStatusResponse.notOk()
                }
            })

        } catch (e: Exception) {
            Toast.makeText(mContext, "خطایی رخ داده است", Toast.LENGTH_SHORT).show()
            serverStatusResponse.notOk()
        }

    }

}
